capstone-backend/src/test/java/com/google/sps/data/GroupTest.java

```diff
@@ -0,0 +1,214 @@
+
+  private static final String ORIGINAL_OWNER = "originalOwner";
+  private static final String NEW_OWNER = "newOwner";
+
+  private static final String MEMBER_ONE = "memberOne";
+  private static final String MEMBER_TWO = "memberTwo";
+  private static final String MEMBER_THREE = "memberThree";
+
+
+  @Before
+  public void setUp() {
+    group = new Club(ORIGINAL_NAME, ORIGINAL_DESCRIPTION, ORIGINAL_OWNER, "");
+  }
+  
+  @Test
+  public void testConstructor() {
+    Assert.assertEquals(group.getName(), ORIGINAL_NAME);
+    Assert.assertEquals(group.getDescription(), ORIGINAL_DESCRIPTION);
+    Assert.assertEquals(group.getAnnouncement(), "");
+    Assert.assertEquals(group.getPosts().size(), 0);
+    Assert.assertEquals(group.getOwnerID(), ORIGINAL_OWNER);
+    Assert.assertEquals(group.getMemberIDs().size(), 0);
+    Assert.assertEquals(group.getInviteIDs().size(), 0);
+  }
+
+  @Test 
+  public void testName() {
+    group.setName(NEW_NAME);
+    Assert.assertEquals(group.getName(), NEW_NAME);
+  }
+
+  @Test 
+  public void testDescription() {
+    group.setDescription(NEW_DESCRIPTION);
+    Assert.assertEquals(group.getDescription(), NEW_DESCRIPTION);
+  }
+
+  @Test 
+  public void testAnnouncement() {
+    group.setAnnouncement(ANNOUNCEMENT);
+    Assert.assertEquals(group.getAnnouncement(), ANNOUNCEMENT);
+  }
+
+  @Test
+  public void testPost() {
+    group.post(POST_ONE);
+
+    Assert.assertEquals(group.getPosts().size(), 1);
+
+    List<String> list = new ArrayList<>(group.getPosts());
+    Assert.assertEquals(list.get(0), POST_ONE);
+
+    group.post(POST_TWO);
+
+    Assert.assertEquals(group.getPosts().size(), 2);
+    list = new ArrayList<>(group.getPosts());
+    Assert.assertEquals(list.get(0), POST_ONE);
+    Assert.assertEquals(list.get(1), POST_TWO);
+  }
+
+  @Test
+  public void testClearPost() {
+    group.post(POST_ONE);
+    group.post(POST_TWO);
+
+    Assert.assertEquals(group.getPosts().size(), 2);
+
+    group.clearPosts();
+
+    Assert.assertEquals(group.getPosts().size(), 0);
+  }
+
+  @Test 
+  public void testOwner() {
+    group.setOwnerID(NEW_OWNER);
+    Assert.assertEquals(group.getOwnerID(), NEW_OWNER);
+  }
+
+  @Test
+  public void testInvite() {
+    group.invite(MEMBER_ONE);
+
+    Assert.assertEquals(group.getInviteIDs().size(), 1);
+    List<String> list = group.getInviteIDs();
+    Assert.assertTrue(list.contains(MEMBER_ONE));
+
+    group.invite(MEMBER_TWO);
+
+    Assert.assertEquals(group.getInviteIDs().size(), 2);
+    list = group.getInviteIDs();
+    Assert.assertTrue(list.contains(MEMBER_ONE));
+    Assert.assertTrue(list.contains(MEMBER_TWO));
+  }
+
+  @Test
+  public void testInviteList() {
+    group.invite(Arrays.asList(MEMBER_ONE, MEMBER_TWO));
+
+    Assert.assertEquals(group.getInviteIDs().size(), 2);
+    List<String> list = group.getInviteIDs();
+    Assert.assertTrue(list.contains(MEMBER_ONE));
+    Assert.assertTrue(list.contains(MEMBER_TWO));
```

Comment:
Have you tried using `assertEquals` on lists before, e.g.: `Assert.assertEquals(list, Arrays.asList(MEMBER_ONE, MEMBER_TWO));`?

Reply:
Yup this works! We were a bit concerned about doing an exact match since collections are unordered in Datastore, but I believe that now they are ordered since we are using Firestore.